pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -255,10 +266,22 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 	var workFeatureGates []operatorapiv1.FeatureGate
 	if klusterlet.Spec.WorkConfiguration != nil {
 		workFeatureGates = klusterlet.Spec.WorkConfiguration.FeatureGates
+		config.WorkKubeAPIQPS = float32(klusterlet.Spec.WorkConfiguration.KubeAPIQPS)
+		config.WorkKubeAPIBurst = klusterlet.Spec.WorkConfiguration.KubeAPIBurst
 	}
 	config.WorkFeatureGates, workFeatureMsgs = helpers.ConvertToFeatureGateFlags("Work", workFeatureGates, ocmfeature.DefaultSpokeWorkFeatureGates)
 	meta.SetStatusCondition(&klusterlet.Status.Conditions, helpers.BuildFeatureCondition(registrationFeatureMsgs, workFeatureMsgs))
 
+	// for singleton agent, the QPS and Burst use the max one between the configurations of registration and work
+	config.AgentKubeAPIQPS = config.RegistrationKubeAPIQPS
+	if config.AgentKubeAPIQPS < config.WorkKubeAPIQPS {
+		config.AgentKubeAPIQPS = config.WorkKubeAPIQPS
+	}
+	config.AgentKubeAPIBurst = config.RegistrationKubeAPIBurst
```

Comment:
seems not correct here. Why set two times?

Reply:
RegistrationKubeAPIQPS/Burst is used in the registration deployment.
WorkKubeAPIQPS/Burst is used in the work deployment.
AgentKubeAPIQPS/Burst is used in the singleton agent, which is to set the max one.